usr/sbin/registercloudguest

```diff
@@ -454,8 +454,11 @@ if args.user_smt_ip or args.user_smt_fqdn or args.user_smt_fp:
         print(msg, file=sys.stderr)
         sys.exit(1)
 
+    ip_addr_version = 'IPv4'
     try:
-        ipaddress.ip_address(args.user_smt_ip)
+        ip_addr = ipaddress.ip_address(args.user_smt_ip)
```

Comment:
Instead of introducing two new variables here:
-  `ip_addr_version`
- `ip_addr`

consider keep the code as it is and use `version` [1] method

```python3
def has_network_access_by_ip_address(server_ip):
    """Check if we can connect to the given server"""
    try:
        connection = socket.create_connection((server_ip, 443), timeout=2)
    except OSError:
        logging.info(
            'Skipping IPv%s protocol version, no network configuration' % ipaddress.ip_address(server_ip).version
            )
        return False

    connection.close()
    return True
```

i.e., using test IP values:
```bash
>>> foo = ipaddress.ip_address('8.8.8.8')
>>> foo.version
4
>>> foo = ipaddress.ip_address('2001:4860:4860::8888')
>>> foo.version
6
```
[1] https://docs.python.org/3/library/ipaddress.html#ipaddress.IPv4Address.version

Reply:
Yeah I considered putting the detection into the `has_network_access_by_ip_address` function then decided against it. I'll think about it some more, it would minimize the overall change set and we already depend on the ipaddress module in the code base as such adding it to this file as a dependency is not really that big of a deal.

Thanks for pointing this out.

Reply:
Pushed the IP protocol version check into `has_network_access_by_ip_address`